astroid/node_classes.py

```diff
@@ -36,13 +36,20 @@
 import itertools
 import pprint
 import sys
+import typing
```

Comment:
I needed to avoid name conflicts as `astroid` itself has a `List` class (in `node_classes.py`). It would have been possible to use `Optional`, but I figured to it's easier to understand if I stick to `typing.xxx` naming. Once we drop support for `3.6` we can simplify it with `from __future__ import annotations` and `list`.